mysql-exporter/mysql-exporter.libsonnet

```
@@ -3,38 +3,61 @@ local container = k.core.v1.container;
 local service = k.core.v1.service;
 local containerPort = k.core.v1.containerPort;
 local deployment = k.apps.v1.deployment;
+local envVar = k.core.v1.envVar;
+
+
+local mysql_credential(config) =
+  if std.length(config.mysql_password) > 0 then
+    [{ name: 'MYSQL_PASSWORD', value: config.mysql_password }]
+  else [envVar.fromSecretRef('MYSQL_PASSWORD', config.mysql_password_secret, config.mysql_password_secret_key)];
+
+
+local mysql_host(config, fqdn) =
+  if std.length(fqdn) > 0 then
+    '%s' % fqdn
+  else
+    '%s.%s.svc.cluster.local' % [
+      config.deployment_name,
+      config.namespace,
+    ];
 
 {
+  image:: 'prom/mysqld-exporter:v0.12.1',
+  mysql_fqdn:: '',
```

```suggestion
  mysql_fqdn:: '%s.%s.svc.cluster.local' %
               [
                 $._config.deployment_name,
                 $._config.namespace,
               ],
```
This gets rid of the `mysql_host()` function, if `$.mysql_fqdn is set by the use, this will be overridden and above errors will simply be ignored.